storage/src/archive/storage.rs

```diff
@@ -227,14 +227,9 @@ impl<T: Translator, E: Storage + Metrics, K: Array, VC: CodecConfig + Copy, V: C
         // Store interval
         self.intervals.insert(index..=index);
 
-        // Store item
-        self.keys.insert(&key, index);
-
-        // Cleanup tracked keys
-        //
-        // We call this after insertion to avoid unnecessary underlying map
-        // operations.
-        self.keys.remove(&key, |index| *index < oldest_allowed);
+        // Insert and prune any useless keys
+        self.keys
```

Comment:
This is a performance improvement (may have been possible in the old code by clear here).